x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/delete_domain_panel.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import { EuiButton, EuiSpacer, EuiText } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { CrawlerSingleDomainLogic } from '../crawler_single_domain_logic';
+
+export const DeleteDomainPanel: React.FC = ({}) => {
+  const { domain } = useValues(CrawlerSingleDomainLogic);
+  const { deleteDomain } = useActions(CrawlerSingleDomainLogic);
+
+  if (!domain) {
+    return null;
+  }
+
+  return (
+    <>
+      <EuiText size="s">
+        <p>
+          <FormattedMessage
+            id="xpack.enterpriseSearch.appSearch.crawler.deleteDomainPanel.description"
+            defaultMessage="Remove this domain from your crawler. This will also delete all entry points and crawl
+                    rules you have setup. {cannotUndoMessage}."
+            values={{
+              cannotUndoMessage: (
+                <strong>
+                  {i18n.translate(
+                    'xpack.enterpriseSearch.appSearch.crawler.deleteDomainPanel.cannotUndoMessage',
+                    {
+                      defaultMessage: 'This cannot be undone',
+                    }
+                  )}
+                </strong>
+              ),
+            }}
+          />
+        </p>
+      </EuiText>
+      <EuiSpacer size="m" />
+      <EuiButton
+        color="danger"
+        iconType="trash"
+        onClick={() => {
+          if (
+            confirm(
+              i18n.translate(
+                'xpack.enterpriseSearch.appSearch.crawler.deleteDomainPanel.confirmationPopupMessage',
+                {
+                  defaultMessage:
+                    "Are you sure you want to remove the domain {domainUrl} from this engine's crawler?",
+                  values: { domainUrl: domain.url },
```

Comment:
Should this also be extracted to utils like `getDeleteDomainSuccessMessage`?

Reply:
I've done this in 83df671f81a